hack/lib/golang.sh

```diff
@@ -610,16 +610,12 @@ kube::golang::build_binaries() {
     local -a targets=()
     local arg
 
-    # Add any files with those //generate annotations in the array below.
-    readonly BINDATAS=( "${KUBE_ROOT}/test/e2e/framework/gobindata_util.go" )
-    kube::log::status "Generating bindata:" "${BINDATAS[@]}"
-    for bindata in ${BINDATAS[@]}; do
-          # Only try to generate bindata if the file exists, since in some cases
-          # one-off builds of individual directories may exclude some files.
-      if [[ -f $bindata ]]; then
-          go generate "${bindata}"
-      fi
-    done
+    # TODO generalize this optimization somehow: Build bindata stuff is only really required for e2e as of now.
+    if [[ $@ == *"e2e"* ]]; then
+	 pushd `dirname test/e2e/`
+	 go generate ./...
```

Comment:
can we maybe instead make this a rule in `Makefile.generated_files`? may as well let `make` do some of the work for us.